pkg/network/tracer/tracer_usm_linux_test.go

```diff
@@ -1059,6 +1095,137 @@ func testHTTPsGoTLSCaptureAlreadyRunningContainer(t *testing.T, cfg *config.Conf
 	checkRequests(t, tr, expectedOccurrences, reqs)
 }
 
+// TLS classification tests
+func TestTLSClassification(t *testing.T) {
+	log.SetupLogger(seelog.Default, "debug")
+
+	cfg := testConfig()
+	cfg.ProtocolClassificationEnabled = true
+	cfg.CollectTCPConns = true
+
+	if !classificationSupported(cfg) {
+		t.Skip("TLS classification platform not supported")
+	}
+
+	// testContext shares the context of a given test.
+	// It contains common variable used by all tests, and allows extending the context dynamically by setting more
+	// attributes to the `extras` map.
+	type testContext struct {
+		// A dynamic map that allows extending the context easily between phases of the test.
+		extras map[string]interface{}
+	}
+
+	type tlsTest struct {
+		name            string
+		context         testContext
+		preTracerSetup  func(t *testing.T, ctx testContext)
+		postTracerSetup func(t *testing.T, ctx testContext)
+		validation      func(t *testing.T, ctx testContext, tr *Tracer)
+		teardown        func(t *testing.T, ctx testContext)
+	}
+	tests := []tlsTest{}
+	for _, tlsVersion := range []string{"-tls1", "-tls1_1", "-tls1_2", "-tls1_3"} {
+		tests = append(tests, tlsTest{
+			name: "TLS" + tlsVersion + "_docker",
+			context: testContext{
+				extras: make(map[string]interface{}),
+			},
+			preTracerSetup: func(t *testing.T, ctx testContext) {
+			},
+			postTracerSetup: func(t *testing.T, ctx testContext) {
+				clientSuccess := false
+				var wg sync.WaitGroup
+				wg.Add(1)
+				go func() {
+					time.Sleep(5 * time.Second)
+					clientSuccess = prototls.RunClientOpenssl(t, "localhost", "44330", tlsVersion)
+					wg.Done()
+				}()
+				prototls.RunServerOpenssl(t, "44330", "-www")
+				wg.Wait()
+				if !clientSuccess {
+					t.Fatalf("openssl client failed")
+				}
+			},
+			validation: func(t *testing.T, ctx testContext, tr *Tracer) {
+				// Iterate through active connections until we find connection created above
+				require.Eventuallyf(t, func() bool {
+					payload := getConnections(t, tr)
+					for _, c := range payload.Conns {
+						if c.DPort == 44330 && isTLSTag(c.StaticTags) {
+							return true
+						}
+					}
+					return false
+				}, 4*time.Second, time.Second, "couldn't find TLS connection matching: dstport %d", 44330)
```

Comment:
nit: (unless we parameterized the port)
```suggestion
				}, 4*time.Second, time.Second, "couldn't find TLS connection matching: dstport 44330")
```

Reply:
done